federation/planner_helpers.go

```diff
@@ -0,0 +1,162 @@
+package federation
+
+import "github.com/samsarahq/thunder/graphql"
+
+/*
+-----------------------------------------
+Begin: Recursive Selection Set Generation
+-----------------------------------------
+This code is used to recursively generate a selection set for an object. The purpose is to allow for passing entire object (not scalar)
+fields from team gql service -> team gql service. This can be used when the object fetch from db is too complicated/expensive and its easier
+to simply pass the object rather than refetch it from keys.
+Example:
+Let's say we have object Devices that has been federated:
+		type Devices struct
+		{
+			Id int | key
+			Location LocationInput | key
```

Comment:
I think this means that `Location` also needs to be defined as a GraphQL object with `FetchObjectFromKeys`.
I'd like to call this out somewhere in the documentation.

Reply:
done